dpppt-backend-sdk/dpppt-backend-sdk-interops/README.md

```diff
@@ -0,0 +1,153 @@
+
+The Interoperability Service therefore implements the following batch logic:
+
+#### Download 
+
+1) get `batchTag` of last successfully downloaded batch from `today - retentationDay` (this is needed, as we cannot know if we downloaded the last batch, as it could be that the Interoperability Services was shut down over a certain period of time)
+
+2) use `batchTag` to download the next available batch.
+
+3) if the `batchTag` of the newly downloaded batch is not empty, and the batch itself is non empty, insert it into the database
+
+4) check if the `nextBatchTag` is not null and goto `2)`
+
+5) increase the `downloadDate` by 1 and repeat `2)` through `4)` (until the last batch of today is downloaded)
+
+### Uploading Batching Logic
+
+During upload the EFGS documentation specifies that a batch _MUST NOT_ exceed 4000 keys. Hence, if more keys are available, the set of keys has to be partitioned. Each of those batches a unique batch tag has to be assigned. 
+#### Upload
+
+1) get all keys which are available for upload
+
+2) partition keys into subsets with a size of maximal `MAX_UPLOAD_BATCH_SIZE`
+
+3) for each partition assign a unique `batchTag` and upload to EFGS
+## Days Since Onset of Symptoms
+
+Since the EN framework does not provide a way to encode meta information about the infection of someone who uploaded the key. This though can be crucial (e.g.symptomatic vs asymptomatic, or fixed onset vs ranged onset) in certain situations to estimate the risk factor on the client.
+
+In order to solve this flaw of the EN framework, the EFGS introduced an encoding of such meta information into the `days_since_onset_of_symptoms` field of the EN struct. The following is an overview of how the encoding works, and explained in context of the `EfgsDsosFilter` class used in the Swiss instance.
+
+The encoding uses mutually exclusive ranges over the whole domain of possible values ( the `days_since_onset_of_symptoms` field is a signed 32-bit integer) to encode different ways of explaining the circumstances of the onset of symptoms and the date. 
+### Symptomatic with known onset
+`Range: [-14,+14]`
+
+`0: Onset coincides with today's key date`
+
+User knows exactly the Date of onset of Symptoms (e.g. cough started 5 days ago before submission)
+### Symptomatic with onset range of `n` days
+`Range: [n*100-14,n*100+14]`
+
+`n*100: Onset coincides with yesterday's key date`
+
+User knows just a time range. (e.g. cough started within the last week)
+
+### Symptomatic with unknown onset (days since submission)
+`Range: [2000-14,2000+14] = [1986, 2014]`
+
+`2000: The onset coincides with this keys' date`
+
+User knows that he had symptoms, but didn't know exactly when they started. 
+
+### Asymptomatic (days since submission)
+`Range: [3000-14,3000+14] = [2986, 3014]`
+
+`3000: The onset coincides with this keys' date`
+
+User knows that he had definitely nothing (e.g. never had cough, fever etc.)
```

Comment:
rephrase. "user knows that he definitely had no symptoms". current phrasing suggests he didn't have covid

Reply:
:1